lib/db/mysql.js

```diff
@@ -752,6 +753,66 @@ module.exports = function (log, error) {
   }
 
 
+  // USER EMAILS
+  // Insert : emails
+  // Values : normalizedEmail = $1, email = $2, uid = $3, emailCode = $4, isVerified = $5, isPrimary = $6, verifiedAt = $7, createdAt = $8
+  var CREATE_EMAIL = 'CALL createEmail_1(?, ?, ?, ?, ?, ?, ?, ?)'
+  MySql.prototype.createEmail = function (uid, data) {
+    return this.write(
+      CREATE_EMAIL,
+      [
+        data.normalizedEmail,
+        data.email,
+        uid,
+        data.emailCode,
+        data.isVerified,
+        data.isPrimary,
+        data.verifiedAt,
+        data.createdAt
+      ]
+    )
+      .catch(function(err){
+        // Signal exception is triggered when an attempt to
+        // add an email that is in the users table already.
+        if (err.errno === ER_SIGNAL_EXCEPTION) {
```

Comment:
I believe `ER_DUP_ENTRY` gets called when you are writing the same record to the table. In this case, I want to ensure that you can not write an email in the users table into the email table. I added some tests cases for this and they seem to work as expected.